analyses/cell-type-neuroblastoma-04/exploratory-notebooks/scanvi-singler-comparison.Rmd

```diff
@@ -0,0 +1,391 @@
+  facet_wrap(vars(celltype_method)) +
+  coord_equal() +
+  theme(
+    legend.position = "bottom", 
+    axis.text = element_blank(), 
+    axis.ticks = element_blank()
+  ) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1)))
+```
+
+
+## Heatmap
+
+This section shows a stacked heatmap comparing the `SingleR` and `scANVI` annotations to the consensus validation groups.
+
+```{r, fig.height = 13, fig.width = 12} 
+# make a wide version for the heatmap
+celltype_wide_df <- celltype_df |>
+  dplyr::select(-label) |>
+  tidyr::pivot_wider(
+    names_from = celltype_method,
+    values_from = label_recoded
+  )
+
+list(
+  "SingleR annotations" = make_jaccard_matrix(celltype_wide_df, "validation_group_annotation", "singler"),
+  "scANVI annotations" = make_jaccard_matrix(celltype_wide_df, "validation_group_annotation", "scanvi")
+) |>
+  purrr::imap(
+    \(jaccard_mat, name) create_single_heatmap(
+        mat = jaccard_mat,
+        row_title = name,
+        column_title = "Consensus validation groups",
+        keep_legend_name = "SingleR annotations" # arbitrary
+    )
+  ) |>
+  # concatenate vertically into HeatmapList object
+  purrr::reduce(ComplexHeatmap::`%v%`) |>
+  ComplexHeatmap::draw(heatmap_legend_side = "right")
+```
+
+
+We'll also show some statistics from a confusion matrix comparison between `SingleR` and `scANVI`:
+
+```{r}
+# ensure factors are compatible for caret
+all_levels <- c(
+  celltype_wide_df$scanvi, 
+  celltype_wide_df$singler
+) |> unique()
+
+caret::confusionMatrix(
+  factor(celltype_wide_df$scanvi, levels = all_levels),  
+  factor(celltype_wide_df$singler, levels = all_levels)
+)$overall
+
+```
+
+## Cross-library comparison
+
+Across libraries, what fraction of `SingleR` and `scANVI` labels exactly agree?
+Note that this analysis does not consider similar labels (e.g. two types of T cells) but only does a quick look at identical labels.
+We'll also highlight the PDX samples here to see if there was any associated bias.
+
+
+```{r, fig.width = 10, fig.height = 4}
+frac_agree_df <- celltype_wide_df |>
+  tidyr::separate(cell_id, into = c("library_id", "barcode")) |>
+  dplyr::group_by(library_id) |>
+  dplyr::summarize(
+    frac_agree = sum(scanvi == singler)/dplyr::n(), 
+    library_size = dplyr::n()
+  )
+frac_agree_df <- colData(merged_sce) |> 
+  as.data.frame() |> 
+  dplyr::select(library_id, is_xenograft) |> 
+  unique() |>
+  dplyr::right_join(frac_agree_df, by = "library_id")
+
+
+p1 <- ggplot(frac_agree_df) + 
+  aes(x = frac_agree) + 
+  geom_histogram() + 
+  xlab("Fraction of SingleR and scANVI cells\nthat agree, per library")
+
+p2 <- ggplot(frac_agree_df) + 
+  aes(x = library_size, y = frac_agree, color = is_xenograft) + 
+  geom_point() +
+  labs(
+    x = "Library size (n cells)",
+    y = "Fraction of SingleR and scANVI\ncells that agree"
+  ) +
+  theme(legend.position = "bottom")
+
+p1 + p2
+
+```
+
+## Conclusions
```

Comment:
I think it would be helpful to get a table in here that shows you % of each library that has exact agreement between `SingleR` and `scANVI`, % of each library where cells are in the same family (e.g., the same cell is labeled as two different T cell types), and % of each library that disagrees and is not in the same family.